src/program-spec.ts

```diff
@@ -531,6 +536,99 @@ function ifThenElse<Data>(
   return { type: 'ifThenElse', condition, thenNode, elseNode };
 }
 
+// helpers to create the context
+
+type ContextType = 'zk-app' | 'https';
```

Comment:
can you move the context-related code in this file to its own file? could be `context.ts`. it doesn't really fit here with the spec stuff IMO